### DISCUSSION TITLE
Use-case: publishing npm packages

### DISCUSSION THREAD

**Discussion by armanbilge** [VirtusLab/scala-cli#531] [Show and tell]
Here's a simple project demonstrating how to setup and publish a Scala.js app to npm using scala-cli.
https://github.com/armanbilge/http4s-cli
https://www.npmjs.com/package/http4s-cli

How scala-cli can better support this use-case:
* implement the `jsHeader` option to add the `#!/usr/bin/env node` to the beginning of the script
  https://github.com/scala-js/scala-js/pull/4597
* accept Scala.js options as `using` directives
* release ASAP after a Scala.js minor version bump

**Comment by romanowski** [comment]
Hello @armanbilge, thank you for sharing your use case. 

We have an issue for missing (where I also included `jsHeader`) scala-js options https://github.com/VirtusLab/scala-cli/issues/332 .

> release ASAP after a Scala.js minor version bump

We would like to avoid syncing our releases with other pieces in the Scala ecosystem including Scala.js. Most likely, we will need to use / develop some kind of stable API over scala-js, similar how we [integrate with Scala Native](https://github.com/VirtusLab/scala-cli/pull/449).

**Reply by armanbilge** [reply]
Thanks. That sounds like a much better strategy for SJS updates. mdoc currently has difficulties with this too.

- https://github.com/scalameta/mdoc/issues/601

**Comment by armanbilge** [comment]
Update on this: I just published a new version of [http4s-cli](https://github.com/armanbilge/http4s-cli) against scala-cli 0.1.1. All the updates/changes in scala-cli.js are fantastic and I was able to cleanup my build and remove the hacks. Thank you very much!
